insert_attachments_id_token_btn_plugin/src/Extension/InsertAttachmentsIdToken.php

```diff
@@ -0,0 +1,137 @@
+use Joomla\Database\DatabaseDriver;
+use Joomla\Event\SubscriberInterface;
+
+// no direct access
+defined( '_JEXEC' ) or die('Restricted access');
+
+/**
+ * Button that allows you to add attachments from the editor
+ *
+ * @package Attachments
+ */
+class InsertAttachmentsIdToken extends CMSPlugin implements SubscriberInterface
+{
+	/**
+	 * $db and $app are loaded on instantiation
+	 */
+	protected ?DatabaseDriver $db = null;
+	protected ?CMSApplication $app = null;
+	protected $parent_type =null;
+	protected $row = null;
+
+	/**
+	 * Load the language file on instantiation
+	 *
+	 * @var    boolean
+	 */
+	protected $autoloadLanguage = true;
+
+	/**
+	 * Returns an array of events this subscriber will listen to.
+	 *
+	 * @return  array
+	 */
+	public static function getSubscribedEvents(): array
+	{
+		return ['onEditorButtonsSetup' => 'onEditorButtonsSetup'];
+	}
+
+
+
+	public function onEditorButtonsSetup(EditorButtonsSetupEvent $event)
+	{
+		$subject  = $event->getButtonsRegistry();
+		$disabled = $event->getDisabledButtons();
+
+		if (\in_array($this->_name, $disabled)) {
+			return;
+		}
+
+		$this->loadLanguage();
+		$app = Factory::getApplication();
+		$jinput = $app->getInput();
+		$this->editor = $jinput->getString('editor', null);
+		$this->id = $jinput->getInt('id', null);
+		$button = $this->onDisplay($event->getEditorId(), $jinput->getInt('id', null));
+
+		if ($button && $this->_name) {
+			$subject->add(new Button($this->_name, $button->getProperties()));
+		}
+	}
+
+	/**
+	 * Add Attachment button
+	 *
+	 * @param string $name The name of the editor form
+	 * @param int $asset The asset ID for the entity being edited
+	 * @param int $author The ID of the author of the entity
+	 *
+	 * @return CMSObject button
+	 */
+	public function onDisplay($name, $id)
+	{
+		// Add the regular css file
+		HTMLHelper::stylesheet('media/com_attachments/css/attachments_list.css');
+		HTMLHelper::stylesheet('media/com_attachments/css/attachments_list_dark.css');
+		HTMLHelper::stylesheet('media/com_attachments/css/add_attachment_button.css');
+
+		// Handle RTL styling (if necessary)
+		$lang = $this->app->getLanguage();
+				// Get ready for language things
+		$lang =	 $this->app->getLanguage();
+		if ( !$lang->load('plg_editors-xtd_insert_attachments_id_token', dirname(__FILE__)) ) {
+			// If the desired translation is not available, at least load the English
+			$lang->load('plg_editors-xtd_insert_attachments_id_token', JPATH_ADMINISTRATOR, 'en-GB');
+			}
+		if ( $lang->isRTL() ) {
+			HTMLHelper::stylesheet('media/com_attachments/css/attachments_list_rtl.css');
+			HTMLHelper::stylesheet('media/com_attachments/css/add_attachment_button_rtl.css');
+			}
+
+		// Load the language file from the frontend
+		$lang->load('com_attachments', JPATH_ADMINISTRATOR.'/components/com_attachments');
+
+		$link = "/index.php?option=com_attachments&tmpl=component&parent_id=" . $id;
+		$link .= '&amp;editor=' . $name . '&amp;XDEBUG_SESSION_START=test';
```

Comment:
the '&amp;XDEBUG_SESSION_START=test' string should be removed. You should try to use an extension like the [xdebug helper](https://chromewebstore.google.com/detail/xdebug-helper/eadndfjplgieldjbigjakmdgkmoaaaoc) to avoid hard coding the xdebug session name in your code.